ingestr/src/sources.py

```diff
@@ -4461,3 +4461,211 @@ def dlt_source(self, uri: str, table: str, **kwargs):
+            resource_name = "segments"
+            ad_account_id = table.split(":", 1)[1]
+            if not ad_account_id:
+                raise ValueError(
+                    "ad_account_id must be provided in format 'segments:ad_account_id'"
+                )
+        elif table == "segments":
+            # If just "segments" without specific ID, will fetch all ad accounts and their segments
+            resource_name = "segments"
+            ad_account_id = None
+            if not organization_id:
+                raise ValueError(
+                    "organization_id is required for 'segments' table when no specific ad_account_id is provided"
+                )
+        elif not organization_id and table != "organizations":
+            raise ValueError(
+                f"organization_id is required for table '{table}'. Only 'organizations' table does not require organization_id."
+            )
+
+        if resource_name not in self.resources:
+            raise UnsupportedResourceError(table, "Snapchat Ads")
+
+        from ingestr.src.snapchat_ads import snapchat_ads_source
+
+        source_kwargs = {
+            "refresh_token": refresh_token[0],
+            "client_id": client_id[0],
+            "client_secret": client_secret[0],
+        }
+
+        if organization_id:
+            source_kwargs["organization_id"] = organization_id[0]
+
+        # Add interval_start and interval_end for client-side filtering
+        interval_start = kwargs.get("interval_start")
+        if interval_start:
+            source_kwargs["start_date"] = interval_start
+
+        interval_end = kwargs.get("interval_end")
+        if interval_end:
+            source_kwargs["end_date"] = interval_end
+
+        source = snapchat_ads_source(**source_kwargs)
+
+        # Handle invoices specially - it needs ad_account_id parameter
+        if resource_name == "invoices":
+            return source.invoices(ad_account_id)
+
+        # Handle campaigns specially - it needs ad_account_id parameter
+        if resource_name == "campaigns":
+            return source.campaigns(ad_account_id)
+
+        # Handle adsquads specially - it needs ad_account_id parameter
+        if resource_name == "adsquads":
+            return source.adsquads(ad_account_id)
+
+        # Handle ads specially - it needs ad_account_id parameter
+        if resource_name == "ads":
+            return source.ads(ad_account_id)
+
+        # Handle event_details specially - it needs ad_account_id parameter
+        if resource_name == "event_details":
+            return source.event_details(ad_account_id)
+
+        # Handle creatives specially - it needs ad_account_id parameter
+        if resource_name == "creatives":
+            return source.creatives(ad_account_id)
+
+        # Handle segments specially - it needs ad_account_id parameter
+        if resource_name == "segments":
+            return source.segments(ad_account_id)
```

Comment:
why not pass `ad_account_id` to source constructor and eliminate this entire block?